packages/generator-langium/templates/test/test/linking/linking.test.ts

```diff
@@ -0,0 +1,53 @@
+import { afterEach, beforeAll, describe, expect, test } from "<%= JestOrVitest %>";
+import { EmptyFileSystem, type LangiumDocument } from "langium";
+import { expandToString as s} from "langium";
+import { parseHelper } from "langium/test";
+import { create<%= LanguageName %>Services } from "../../src/language/<%= language-id %>-module.js";
+import { Model, isModel } from "../../src/language/generated/ast.js";
+
+let services: ReturnType<typeof create<%= LanguageName %>Services>;
+let parse:    ReturnType<typeof parseHelper<Model>>;
+let document: LangiumDocument<Model> | undefined;
+
+beforeAll(async () => {
+    services = create<%= LanguageName %>Services(EmptyFileSystem);
+    parse = parseHelper<Model>(services.<%= LanguageName %>);
+
+    // activate the following if your linking test requires elements from a built-in library, for example
+    // await services.shared.workspace.WorkspaceManager.initializeWorkspace([]);
+});
```

Comment:
Suggestion: We can simplify this quite a bit. 

```suggestion
const services = create<%= LanguageName %>Services(EmptyFileSystem);
const parse = parseHelper<Model>(services.<%= LanguageName %>);
// activate the following if your linking test requires elements from a built-in library, for example
/* beforeAll(async () => {
    await services.shared.workspace.WorkspaceManager.initializeWorkspace([]);
}); */

Reply:
This is how I used to do it in practice. However, the top-level statements as well as the `describe` lambdas might be executed by IDE feature to produce overviews like following, right?
That's why I used the super correct version here.

![image](https://github.com/eclipse-langium/langium/assets/8513883/cbebdd76-0bb4-408e-80e0-5adea375efa7)

Reply:
I see. And those test controllers don't perform a new import, right? So the services don't get reloaded. That's fine then.